diff --git a/node_modules/@reach/utils/dist/reach-utils.esm.js b/node_modules/@reach/utils/dist/reach-utils.esm.js
index 0970aa0..adb67a4 100644
--- a/node_modules/@reach/utils/dist/reach-utils.esm.js
+++ b/node_modules/@reach/utils/dist/reach-utils.esm.js
@@ -215,7 +215,7 @@ function getElementComputedStyle(element, styleProp) {
  */

 function getOwnerDocument(element) {
-  return canUseDOM() ? element ? element.ownerDocument : document : null;
+  return canUseDOM() ? element ? element.getRootNode() : document : null;
 }
 /**
  * TODO: Remove in 1.0
