diff --git a/node_modules/@wordpress/e2e-test-utils/build/activate-theme.js b/node_modules/@wordpress/e2e-test-utils/build/activate-theme.js
index e171fe9..c1b2632 100644
--- a/node_modules/@wordpress/e2e-test-utils/build/activate-theme.js
+++ b/node_modules/@wordpress/e2e-test-utils/build/activate-theme.js
@@ -25,6 +25,10 @@ async function activateTheme(slug) {
     return;
   }
   await page.click(`div[data-slug="${slug}"] .button.activate`);
+
+  // Switch to themes.php page as some themes can redirect to a different page after activation.
+  await (0, _visitAdminPage.visitAdminPage)('themes.php');
+
   await page.waitForSelector(`div[data-slug="${slug}"].active`);
   await (0, _switchUserToTest.switchUserToTest)();
 }
