Index: lib/Sabberworm/CSS/Property/Selector.php
===================================================================
--- lib/Sabberworm/CSS/Property/Selector.php	(revision 134f4e62fe8ab9f316425f3c0f480b3f4f52d804)
+++ lib/Sabberworm/CSS/Property/Selector.php	(date 1581832944211)
@@ -41,12 +41,12 @@
 	private $iSpecificity;

 	public static function isValid($sSelector) {
-		return preg_match("/^([a-zA-Z0-9\x{00A0}-\x{FFFF}_\^\$\|\*\=\"\'\~\[\]\(\)\-\s\.:#\+\>]*|\s*?[\+-]?\d+\%\s*)$/u", $sSelector);
+		return preg_match("/^([a-zA-Z0-9,@\\\x{00A0}-\x{FFFF}_\^\$\|\*\=\"\'\~\[\]\(\)\-\s\.:#\+\>]*|\s*?[\+-]?\d+\%\s*)$/u", $sSelector);
 	}

 	public function __construct($sSelector, $bCalculateSpecificity = false) {
 		if (!Selector::isValid($sSelector)) {
-			throw new UnexpectedTokenException("Selector did not match '/^([a-zA-Z0-9\x{00A0}-\x{FFFF}_\^\$\|\*\=\"\'\~\[\]\(\)\-\s\.:#\+\>]*|\s*?[\+-]?\d+\%\s*)$/u'.", $sSelector, "custom");
+			throw new UnexpectedTokenException("Selector did not match '/^([a-zA-Z0-9,@\\\x{00A0}-\x{FFFF}_\^\$\|\*\=\"\'\~\[\]\(\)\-\s\.:#\+\>]*|\s*?[\+-]?\d+\%\s*)$/u'.", $sSelector, "custom");
 		}
 		$this->setSelector($sSelector);
 		if ($bCalculateSpecificity) {
